compendium/DeclarativeServices/test/TestUpdateConfiguration.cpp

```diff
@@ -0,0 +1,517 @@
+/*=============================================================================
+
+  Library: CppMicroServices
+
+  Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+  file at the top-level directory of this distribution and at
+  https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+=============================================================================*/
+
+#include "TestFixture.hpp"
+#include "gtest/gtest.h"
+
+#include "cppmicroservices/Constants.h"
+#include "cppmicroservices/ServiceObjects.h"
+
+#include "TestInterfaces/Interfaces.hpp"
+
+namespace test {
+/**
+   * Verify that a service's configuration can be updated after the service is activated
+   * without deactivating and reactivating the service.
+   */
+TEST_F(tServiceComponent, testUpdateConfig_Modified) //DS_CAI_FTC_1
```

Comment:
I don't see how this test guarantees that the service component wasn't deactivated and activated again when updating the configuration.

Reply:
Line 89 checks that following the Update operation the component is still in the ACTIVE state. Since immediate=false in the manifest.json file, if this component had been deactivated it would be in the SATISFIED state, waiting for a user to do a GetService to request an instance.